src/lib/chips/chip-list.ts

```diff
@@ -49,69 +49,125 @@ import {SPACE, LEFT_ARROW, RIGHT_ARROW} from '../core/keyboard/keycodes';
 })
 export class MdChipList implements AfterContentInit {
 
-  /** Track which chips we're listening to for focus/destruction. */
-  private _subscribed: WeakMap<MdChip, boolean> = new WeakMap();
+  /** Whether or not the chip list is currently focusable via keyboard interaction. */
+  protected _tabIndex = -1;
+
+  /** When a chip is destroyed, we track the index so we can focus the appropriate next chip. */
+  protected _destroyedIndex: number = null;
 
-  /** Whether or not the chip is selectable. */
-  protected _selectable: boolean = true;
+  /** Track which chips we're listening to for focus/destruction. */
+  protected _subscribed: WeakMap<MdChip, boolean> = new WeakMap();
 
   /** The ListKeyManager which handles focus. */
   _keyManager: ListKeyManager;
 
   /** The chip components contained within this chip list. */
   chips: QueryList<MdChip>;
 
-  constructor(private _elementRef: ElementRef) { }
+  constructor(protected _renderer: Renderer, protected _elementRef: ElementRef,
+              @Optional() protected _mdInputContainer: MdInputContainer) {
+  }
 
   ngAfterContentInit(): void {
     this._keyManager = new ListKeyManager(this.chips).withFocusWrap();
 
     // Go ahead and subscribe all of the initial chips
     this._subscribeChips(this.chips);
 
+    // Make sure we set our tab index at the start
+    this._checkTabIndex();
+
     // When the list changes, re-subscribe
     this.chips.changes.subscribe((chips: QueryList<MdChip>) => {
       this._subscribeChips(chips);
+
+      // If the number of chips changes, re-hack any label elements
+      this._hackLabelElement();
+
+      // If we have 0 chips, attempt to focus an input (if available)
+      if (chips.length == 0) {
+        this.focusInput();
+      }
+
+      // Check to see if we need to update our tab index
+      this._checkTabIndex();
+
+      // Check to see if we have a destroyed chip and need to refocus
+      this._checkDestroyedFocus();
     });
   }
 
-  /**
-   * Whether or not this chip is selectable. When a chip is not selectable,
-   * it's selected state is always ignored.
-   */
-  @Input() get selectable(): boolean {
-    return this._selectable;
+  ngAfterViewInit(): void {
+    this._hackLabelElement();
   }
 
-  set selectable(value: boolean) {
-    this._selectable = coerceBooleanProperty(value);
+  /**
+   * Ensure the label element does not contain the md-empty class unless we have 0 chips.
+   *
+   * TODO: Remove when md-input-container allows md-empty configuration.
+   * */
+  protected _hackLabelElement() {
```

Comment:
cc @mmalerba 

What is it that you need the input to do here? I'd rather avoid merging something like this.